### PR TITLE
Implement plotting functionality

### PR DESCRIPTION
**Pull request by busykoala**
Matplotlib has a great interface working together with pandas.
Therefore I chose to use this package for plotting the graphs.

Because we are still using Python 2.7 it was quite a pain to find versions that compile and/or work together.

Matplotlib is using the `Agg` backend because the CI does not support tkinter.

The graphs are plotted and then converted into base64. This allows rendering them within an image tag easily.

### CODE REVIEW

File: ftw/linkchecker/tests/helpers.py [hunk 1]
```diff
@@ -47,6 +55,37 @@ def generate_test_data_excel_workbook():
...
+    with open(TEST_REPORT, 'rb') as f_:
+        f_.seek(0)
+        data = f_.read()
+    file_ = api.content.create(
+        container=file_listing_block, type='File',
+        title=ASSET_NAME, file=data)
+    file_.setFilename(ASSET_NAME)
+    transaction.commit()
+
+    if upload_two:
```
Comment (mbaechtold) [inline ftw/linkchecker/tests/helpers.py:78]: This is just a personal advice (you don't have to change it now): I tend to put refactoring (like moving this function to another module) in a separate commit, before adding new functionality. This makes diffing between the commits much easier, which is useful for debugging.

**Review by mbaechtold — CHANGES_REQUESTED** [review]
Could you please split the first commit into 3 commits:
- Sorting the dependencies
- Moving of production dependencies
- Addition of new dependencies

File: ftw/linkchecker/tests/helpers.py (hunk 1 above)
Comment (busykoala) [reply ftw/linkchecker/tests/helpers.py:78]: I added a separate commit for moving the test function. Thanks for the advice. It makes totally sense to me.

**Comment by busykoala** [comment]
I tried to address the changes requested. I hope this is what you meant.

**Review by mbaechtold — APPROVED** [review]
👍

**Pull request merged**